sdk/storage/azblob/blob/client.go

```diff
@@ -133,6 +133,15 @@ func (b *Client) Undelete(ctx context.Context, o *UndeleteOptions) (UndeleteResp
 	return resp, err
 }
 
+// PermanentDelete permanently deletes soft-deleted snapshots & soft-deleted version blobs and is a dangerous operation and SHOULD NOT BE USED.
+// WARNING: This operation should not be used unless you know exactly the implications. We will not provide support for this API.
+// For more information, see https://docs.microsoft.com/rest/api/storageservices/delete-blob.
+func (b *Client) PermanentDelete(ctx context.Context, o *PermanentDeleteOptions) (PermanentDeleteResponse, error) {
```

Comment:
At blob client level we can have single delete api that takes delete type as input. On service client we can have two apis both converging to same blob client api passing down correct delete type to it?

Reply:
I've updated the code so that DeleteType is an option buried in the DeleteOption bag!